Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -1,48 +1,37 @@
 ﻿using Promact.Trappist.DomainModel.DbContext;
 using System.Linq;
 using Promact.Trappist.DomainModel.Models.Test;
-using System.Collections.Generic;
 using System.Threading.Tasks;
+using System.Collections.Generic;
 using Microsoft.EntityFrameworkCore;
-
+using Promact.Trappist.Utility.GlobalUtil;
 
 namespace Promact.Trappist.Repository.Tests
 {
     public class TestsRepository : ITestsRepository
     {
         private readonly TrappistDbContext _dbContext;
-        public TestsRepository(TrappistDbContext dbContext)
+        private readonly IUtil _util;
+        public TestsRepository(TrappistDbContext dbContext, IUtil util)
         {
             _dbContext = dbContext;
+            _util = util;
         }
-        /// <summary>
-        /// this method is used to create a new test
-        /// </summary>
-        /// <param name="test">object of Test</param>
-        public void CreateTest(Test test)
-        {
+        
+        public async Task CreateTestAsync(Test test)
+        {           
             _dbContext.Test.Add(test);
-            _dbContext.SaveChangesAsync();
-        }
-        /// <summary>
-        /// this method is used to check whether test name is unique or not
-        /// </summary>
-        /// <param name="test">object of Test</param>
-        /// <returns>boolean</returns>
-        public bool UniqueTestName(Test test)
+            test.Link = _util.RandomLinkGenerator(10);
+            await _dbContext.SaveChangesAsync();
+        }                       
+        
+        public async Task<bool> IsTestNameUniqueAsync(string testName)
```

Comment:
IsTestNameUnique should return true if it is unique otherwise false, does your method fulfills that?